features/device-config/rest/src/main/java/org/opennms/features/deviceconfig/rest/impl/DefaultDeviceConfigRestService.java

```diff
@@ -235,9 +244,45 @@ public Response getDeviceConfigsByInterface(Integer ipInterfaceId, String config
 
     /** {@inheritDoc} */
     @Override
-    public void deleteDeviceConfig(long id) {
-        throw new UnsupportedOperationException("Delete device config not yet supported.");
-        // deviceConfigDao.delete(id);
+    public Response deleteDeviceConfig(String id) {
+        if (Strings.isNullOrEmpty(id)) {
+            LOG.debug("deleteDeviceConfig: empty or null id supplied by request");
+            return Response.status(Status.BAD_REQUEST).entity("Invalid 'id' parameter").build();
+        }
+
+        final String idParamPattern = "\\d+(, ?\\d+)*";
+        var pattern = Pattern.compile(idParamPattern);
```

Comment:
Make this static to only compile once.

Reply:
static won't be allowed here, can you elaborate more?